core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQConnection.scala

```diff
@@ -103,7 +113,7 @@ class DefaultRabbitMQConnection[F[_]: Effect](connection: ServerConnection,
     }
   }
 
-  protected def addAutoCloseable[A <: AutoCloseable](a: A): A = {
+  protected def addAutoCloseable[A <: FAutoCloseable[F]](a: F[A]): F[A] = a.map { a =>
     closeablesLock.synchronized {
```

Comment:
I am not an expert but maybe you could use something like `Ref` instead of manual synchronization?

Reply:
I probably could and should do so but I don't have it. It's in cats-effect since 1.0.0 and in FS2 but I don't have none of these dependencies available here...  
Once I'll be able to use cats-effect 1.x, many things will be easier to do.